yb-voyager/cmd/import.go

```
@@ -285,6 +285,10 @@ func validateTargetSchemaFlag() {
 		}
```

@priyanshi-yb do we really need the `if` check on line 282? 
- For import-data-to-source-replica, the same limitation applies, right? 
- For import-data-to-source, there's no way for the user to specify target-db-schema in the first place. so, it should be okay? 

Can we do the tconf.Schema not empty check in all scenarios?

---

> For import-data-to-source-replica, the same limitation applies, right?

yes user should not be allowed to pass schema names and should be same as source

> Can we do the tconf.Schema not empty check in all scenarios?

I think we can't do without importRole because this `validateTargetSchemaFlag()` is run in the commands - `import data` and `import data file`
and for the source-replica there is separate function `validateFFDBSchemaFlag` for this similar flag.

---

so we want this check to run in all scenarios except `import data file` and `import data to source replica`. Does that make sense?

---

yeah and I think that is only import-data then.

---

and import-schema, for which this task was created :)

---

@atharva-kurhade-yb , let's do the following: 
- Remove the new check that you've added, we can re-use the existing check with a couple of modifications
- in the existing check (line 282), 
   - run the check only if `importerRole` is NOT one of (SOURCE_REPLICA_DB_IMPORTER_ROLE, SOURCE_DB_IMPORTER_ROLE, IMPORT_FILE_ROLE)

---

> and import-schema, for which this task was created :)

oh yes, mybad I didn't realize that, I was just thinking about data commands 😅

---

It's definitely a bit weird to rely on importerRole for this because in case of import-schema it will be empty. But I think it might be okay for now.

---

So with this change will it throw an error with import schema instead of a warning?